hbase-server/src/test/java/org/apache/hadoop/hbase/client/FromClientSideBase.java

```diff
@@ -280,6 +282,19 @@ protected Result getSingleScanResult(Table ht, Scan scan) throws IOException {
     return result;
   }
 
+  protected Pair<ScanMetrics, List<ScanMetrics>> getScanMetrics(Table ht, Scan scan)
+    throws IOException {
+    ScanMetrics scanMetrics;
+    List<ScanMetrics> scanMetricsByRegion;
+    try (ResultScanner scanner = ht.getScanner(scan)) {
+      while ((scanner.next()) != null) {
+      }
+      scanMetrics = scanner.getScanMetrics();
+      scanMetricsByRegion = scanner.getScanMetricsByRegion();
+    }
+    return Pair.newPair(scanMetrics, scanMetricsByRegion);
```

Comment:
Yeah, this emphisized to me how it's weird that the new feature is not folded into the existing feature, and is instead a "bolt-on". It's not very nice to use like this.